Makefile.kokkos

```diff
@@ -321,89 +327,106 @@ KOKKOS_LDFLAGS = -L$(shell pwd)
 KOKKOS_SRC =
 KOKKOS_HEADERS =
 
-# Generating the KokkosCore_config.h file.
 
-tmp := $(shell echo "/* ---------------------------------------------" > KokkosCore_config.tmp)
-tmp := $(shell echo "Makefile constructed configuration:" >> KokkosCore_config.tmp)
-tmp := $(shell date >> KokkosCore_config.tmp)
-tmp := $(shell echo "----------------------------------------------*/" >> KokkosCore_config.tmp)
+gen_config=KokkosCore_config.tmp
+gen_header=KokkosCore_config.h
```

Comment:
Proposed name: KOKKOS_CONFIG_HEADER

Reply:
Reason for its not KOKKOS_INTERNAL: other folks may make something directly dependent on this file.